cmd/optimize/events.go

```diff
@@ -460,15 +482,124 @@ func listRecommendations(flags *recommendationsCmdFlags) func(*cobra.Command, []
+			if startedRow, ok := blockerRows[uniqueKey]; !ok {
+				log.Warnf("No optimization_started event found for recommendation with optimizer_id: %v and num: %v", optimizerId, optimizationNum)
+			} else {
+				for attr, val := range startedRow.(map[string]any) {
+					recommendationWithBlockers.BlockersAttributes[attr] = val
+
+					// extract the ID from the attribute string
+					if !strings.Contains(attr, "principal") {
+						splitAttr := strings.Split(attr, ".")
+						if len(splitAttr) > 3 {
+							blockerID := splitAttr[len(splitAttr)-2]
+							if !strings.Contains(strings.Join(recommendationWithBlockers.Blockers, ","), blockerID) {
+								recommendationWithBlockers.Blockers = append(recommendationWithBlockers.Blockers, blockerID)
+							}
+						}
+					}
+				}
+			}
+
+			if len(recommendationWithBlockers.Blockers) > 0 {
+				recommendationWithBlockers.BlockersPresent = "true"
+			}
+
+			recommendationRowsWithBlockers = append(recommendationRowsWithBlockers, recommendationWithBlockers)
+		}
+
 		output.PrintCmdOutput(cmd, struct {
-			Items []eventsRow `json:"items"`
-			Total int         `json:"total"`
-		}{Items: recommendationRows, Total: len(recommendationRows)})
+			Items []recommendationRow `json:"items"`
+			Total int                 `json:"total"`
+		}{Items: recommendationRowsWithBlockers, Total: len(recommendationRowsWithBlockers)})
 
 		return nil
 	}
 }
 
+func getOptimizationBlockerData(tempVals recommendationsTemplateValues) (map[string]any, error) {
+
+	var buff bytes.Buffer
+	if err := optimizationStartedTemplate.Execute(&buff, tempVals); err != nil {
+		return nil, fmt.Errorf("optimizationStartedTemplate.Execute: %w", err)
+	}
+	query := buff.String()
+
+	// execute query, process results
+	resp, err := uql.ExecuteQuery(&uql.Query{Str: query}, uql.ApiVersion1)
+	if err != nil {
+		return nil, fmt.Errorf("uql.ExecuteQuery: %w", err)
+	}
+	if resp.HasErrors() {
+		log.Error("Execution of optimization_started query encountered errors. Returned data may not be complete!")
+		for _, e := range resp.Errors() {
+			log.Errorf("%s: %s", e.Title, e.Detail)
+		}
+	}
+
+	main_data_set := resp.Main()
+	if main_data_set == nil || len(main_data_set.Data) < 1 {
+		return nil, fmt.Errorf("No optimization_started results found for given input\n")
+	}
+	if len(main_data_set.Data[0]) < 1 {
+		return nil, fmt.Errorf("Main dataset %v first row has no columns", main_data_set.Name)
+	}
+
+	data_set, ok := main_data_set.Data[0][0].(*uql.DataSet)
+	if !ok {
+		return nil, fmt.Errorf("Main dataset %v first row first column (type %T) could not be converted to *uql.DataSet", main_data_set.Name, main_data_set.Data[0][0])
+	}
+	startedBlockersData, err := extractStartedBlockersData(data_set)
+	if err != nil {
+		return nil, fmt.Errorf("extractStartedBlockersData: %w", err)
+	}
+
+	return startedBlockersData, nil
+}
+
+func extractStartedBlockersData(dataset *uql.DataSet) (map[string]any, error) {
+	resp_data := &dataset.Data
```

Comment:
Check for nil and return empty, otherwise panics can occur when uql backend has errors